packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -332,10 +332,17 @@ export class RichTextEditor extends FoundationElement {
     }
 
     private updateEditorButtonsState(): void {
+        const { extensionManager, state } = this.tiptapEditor;
+        const { extensions } = extensionManager;
+        const { selection } = state;
+        const parentList = findParentNode(node => isList(node.type.name, extensions))(selection);
+
         this.boldButton.checked = this.tiptapEditor.isActive('bold');
         this.italicsButton.checked = this.tiptapEditor.isActive('italic');
-        this.bulletListButton.checked = this.tiptapEditor.isActive('bulletList');
-        this.numberedListButton.checked = this.tiptapEditor.isActive('orderedList');
+        this.bulletListButton.checked = parentList !== undefined
+            && parentList.node.type.name === 'bulletList';
+        this.numberedListButton.checked = parentList !== undefined
+            && parentList.node.type.name === 'orderedList';
```

Comment:
minor: Move `bulletList` and `orderedList` string literals to a [constant enum](https://github.com/ni/nimble/blob/32c8dd6d1e921581b13eb4ac82d5f0331d323752/packages/nimble-components/docs/coding-conventions.md#use-const-objects-instead-of-typescript-enums) `TipTapNodeNames` for better maintainability later on upgrades.